File: src/problems/dontmatter/47.py
import math

n = int(input())
max_val = -1
max_index = -1

for i in range(n):
    d, c = map(int, input().split())
    val = c * math.log(d)
    if val > max_val:
        max_val = val
        max_index = i

print(max_index)

File: src/problems/dontmatter/47.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, maxIndex int
	var maxVal float64 = -1
	fmt.Scan(&n)

	for i := 0; i < n; i++ {
		var d, c int
		fmt.Scan(&d, &c)
		val := math.Pow(float64(d), float64(c))
		if val > maxVal {
			maxVal = val
			maxIndex = i
		}
	}

	fmt.Println(maxIndex)
}
